packages/preferences/src/browser/preferences-tree-widget.ts

```diff
@@ -488,6 +477,9 @@ export class PreferencesTreeWidget extends TreeWidget {
 
     protected onAfterAttach(msg: Message): void {
         this.initializeModel();
+        this.toDispose.push(this.preferenceSchemaProvider.onDidPreferenceSchemaChanged(() => {
```

Comment:
`toDispose` -> `toDisposeOnDetach`

So when a widget is moved listeners is removed as well.

Reply:
Done and squashed all changes into single commit.